BrowserKit/Sources/Redux/StoreSubscriber.swift

```diff
@@ -11,17 +11,36 @@ public protocol AnyStoreSubscriber: AnyObject {
 
 /// Subscribers listen for state changes.
 /// While all reducers will be called when a action gets dispatched, the subscriber will only be called
-/// when an actual state change happens.
+/// when an actual state change happens. See also: notes below in `newState(state: Any)`
 public protocol StoreSubscriber: AnyStoreSubscriber {
     associatedtype SubscriberStateType
 
+    /// Updates the subscriber with a new State for its screen state type.
+    /// - Parameter state: the changed screen state.
     func newState(state: SubscriberStateType)
+
+    /// Provides a hook that Subscribers may use to pre-filter State updates.
+    /// This is mandatory since it is critical in order to ensure state changes
+    /// for identical Screen types are not incorrectly sent to different windows.
+    /// Before sending `newState()`, the Subscriber will first be asked if it is
+    /// interested in the given State. A Subscriber will typically check whether
+    /// the incoming State has a matching WindowUUID.
+    ///
+    /// Allowing Redux to update the same screen across multiple windows is also
+    /// facilitated by the inverse; Subscribers, if they wish to allow Actions
+    /// on other windows to update their own state, may simply return true here
+    /// and they will receive all screen state updates for their given type.
+    /// - Parameter state: the state update that is incoming.
+    /// - Returns: return true to receive the state update.
+    func validateState(state: SubscriberStateType) -> Bool
```

Comment:
Note to reviewers: you might be wondering why this check couldn't just be performed as part of the `newState()` update. For example, if a Subscriber wants to make sure that they are updating only for their specific iPad window and ignore other windows' States, couldn't they simply compare the UUIDs at the top of `newState()`? The answer is yes they could, the problem is that it creates a hidden contract/requirement that every engineer will have to remember any time they add a new Subscriber to the client.

It will be very easy for folks to forget about that, in practice. If they do forget, there's a good chance their code is going to break during multi-window use. In order to avoid potential future bugs, this new protocol hook was added so that it forces any new Subscribers we write to explicitly declare whether they want to filter State changes by window, or whether to just accept any State change for their associated screen/state type.

This was the best solution I could find for this but as usual I'm happy to discuss or chat about other approaches.